modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+        when(AbstractResourceManager.encodeSCIMException(any(NotFoundException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new NotFoundException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(NotFoundException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateCharonException")
+    public Object[][] dataToTestCreateCharonException() {
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateCharonException")
+    public void testCreateCharonException(String scimObjectString, String attributes,
+                                          String excludeAttributes,
+                                          int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(ConflictException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenThrow(CharonException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestDeleteSuccess")
+    public Object[][] dataToTestDeleteSuccess()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+        String username = user.getUsername();
+        return new Object[][]{
+                {username, ResponseCodeConstants.CODE_NO_CONTENT}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestDeleteSuccess")
+    public void testDeleteSuccess(String userName, int expectedScimResponseStatus)
+            throws NotFoundException {
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+
+        SCIMResponse outputScimResponse = meResourceManager.delete(userName, userManager);
+
+        //Assertions
```

Comment:
in some methods, this comment is there and in others, it is not there. IMO this comment is not required

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d